rocAL/rocAL_pybind/examples/rocAL_api_python_unittest.py

```diff
@@ -52,154 +73,290 @@ def main():
     rocal_cpu = False if args.rocal_gpu else True
     device = "cpu" if rocal_cpu else "cuda"
     batch_size = args.batch_size
+    max_height = args.max_height
+    max_width = args.max_width
+    color_format = types.RGB if args.color_format else types.GRAY
+    tensor_layout = types.NHWC if args.color_format else types.NCHW
+    tensor_dtype = types.UINT8
     num_threads = args.num_threads
     random_seed = args.seed
-    local_rank =  args.local_rank
-    world_size =  args.world_size
-    display = True if args.display else False
+    local_rank = args.local_rank
+    world_size = args.world_size
+    interpolation_type = INTERPOLATION_TYPES[args.resize_interpolation_type]
+    scaling_mode = SCALING_MODES[args.resize_scaling_mode]
+    if (scaling_mode != types.SCALING_MODE_DEFAULT and interpolation_type !=
+            types.LINEAR_INTERPOLATION):
+        interpolation_type = types.LINEAR_INTERPOLATION
+    if augmentation_name in ["hue", "saturation", "color_twist"] and color_format == types.GRAY:
+        print("Not a valid option! Exiting!")
+        sys.exit(0)
+
     try:
-        path= "OUTPUT_IMAGES_PYTHON/NEW_API/FILE_READER/" + args.augmentation_name
+        path = "OUTPUT_IMAGES_PYTHON/NEW_API/FILE_READER/" + args.augmentation_name
         isExist = os.path.exists(path)
         if not isExist:
             os.makedirs(path)
     except OSError as error:
         print(error)
     # Create Pipeline instance
-    pipe = Pipeline(batch_size=batch_size, num_threads=num_threads, device_id=local_rank, seed=random_seed, rocal_cpu=rocal_cpu, tensor_layout=types.NHWC if args.NHWC else types.NCHW , tensor_dtype=types.FLOAT16 if args.fp16 else types.FLOAT)
+    pipe = Pipeline(batch_size=batch_size,
+                    num_threads=num_threads,
+                    device_id=local_rank,
+                    seed=random_seed,
+                    rocal_cpu=rocal_cpu,
+                    tensor_layout=tensor_layout,
+                    tensor_dtype=tensor_dtype,
+                    output_memory_type=types.CPU_MEMORY if rocal_cpu else types.GPU_MEMORY)
```

Comment:
why do we need to pass out_memory_type here. Shouldn't it be implied?
Also I would suggest using HOST/device instead of CPU_MEMORY and GPU_MEMORY

Reply:
The copy_data function in tensor.cpp requires the memory type passed as the second argument. If its not passed, it sets memory type as CPU_MEMORY and assumes the ptr passed is a host buffer ptr and uses memcpy to transfer data. To avoid this, we set the memory type in pipeline and pass the memory type to the copy_data function in C++

Reply:
@SundarRajan28 Please use host and device as mentioned

Reply:
Renamed both types as HOST_MEMORY and DEVICE_MEMORY respectively